### PR TITLE
Add an FAQ page to the Redux docs

### PR DESCRIPTION
**Pull request by markerikson**
Here you go.  I created a new branch and added the completed contents of the FAQ page in a single commit, since I don't think all 49 individual commits really need to go into Redux's permanent history :)

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -0,0 +1,552 @@
+# Redux FAQ
+
+## General
+
+### When should I use Redux?
+
+Pete Hunt, creator of React, says:
```
Comment (gaearon) [inline docs/FAQ.md]: “one of the early contributors to React”

Comment (gaearon) [inline docs/FAQ.md]: It’s best to use Pete’s own words:

>I was one of the original members of the Facebook team that built and open-sourced React.

File: docs/FAQ.md [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+
+#### Further information
+**Documentation**:
+- [Introduction: Motivation](introduction/Motivation.md)
+
+**Discussions**:
+- [React How-To](https://github.com/petehunt/react-howto)
+- [Twitter - Don't use Redux until...](https://twitter.com/dan_abramov/status/699241546248536064)
+- [The Case for Flux](https://medium.com/swlh/the-case-for-flux-379b7d1982c6)
+- [SO - Why use Redux over Facebook Flux?](http://stackoverflow.com/questions/32461229/why-use-redux-over-facebook-flux)
```
Comment (gaearon) [inline docs/FAQ.md]: “SO” is a bit confusing. Let’s say StackOverflow everywhere?

File: docs/FAQ.md [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+>Nope, it's just HTML, some artisanal `<script>` tags, and plain old DOM manipulation. Enjoy!
+
+
+## Reducers
+
+### How do I share state between two reducers? Do I have to use `combineReducers`?
+
+The suggested structure for a Redux store is to split the state object into multiple "slices" or "domains" by key, and provide a separate reducer function to manage each individual data slice.  This is similar to how the standard Flux pattern has multiple independent stores, and Redux provides the [`combineReducers`](api/combineReducers.md) utility function to make this pattern easier.  However, it's important to note that `combineReducers` is NOT required - it is simply a utility function for the common use case of having a single reducer function per state slice, with plain Javascript objects for the data.
+
+Many users later want to try to share data between two reducers, but find that `combineReducers` does not allow them to do so.  There are several approaches that can be used:
```
Comment (Aweary) [inline docs/FAQ.md]: You've got an extra space before "There".

**Comment by markerikson** [comment]
Any other comments or concerns besides those?

**Comment by samsch** [comment]
It might be nice to have a linked table of contents at the top. Not sure it's something that should stop the initial pull though.

**Comment by markerikson** [comment]
I can make the "SO" -> "Stack Overflow" and "creator of React" edits easily enough, I've just been waiting to see if you have any other comments or concerns so I can address them all at once.  I know you've been busy, but can you let me know if you have any further changes I should make?  Thanks!

**Comment by markerikson** [comment]
Hi, @gaearon .  Went ahead and made those suggested tweaks. Anything else you'd like to see right now, or is it ready to go ahead and merge?  I'd really like to get this in so I can start pointing people to the generated docs page rather than my own repo fork.

**Comment by mxstbr** [comment]
This is 👌, well done @markerikson!

**Comment by markerikson** [comment]
@samsch : I'm totally fine with potential edits (TOC, maybe splitting into sub-pages, etc).  Just want to get the initial version merged in first, and then it's totally up for improvements by anyone who wants to submit them.

**Comment by mxstbr** [comment]
Lets merge this and iterate on it with PRs!

**Comment by markerikson** [comment]
Sweet.  Does someone have to re-publish the docs for this to show up?

**Comment by mxstbr** [comment]
One second, I'll see if I can do it...

**Comment by mxstbr** [comment]
:tada: http://redux.js.org/docs/FAQ.html :tada:

**Comment by markerikson** [comment]
Boo.  Yah.

@gaearon , @mxstbr , thank you very much.

**Comment by gaearon** [comment]
Amazing stuff!

**Pull request merged**